### ISSUE TITLE
Add Custom W&B Logging from Tempest

### ISSUE THREAD

**Issue by siddk** [stanford-mercury/transformers#2]
Borrow metric and memory usage trackers from Tempest. Reimplement so conducive with HF.Trainer().
